targets/liteos-stm32f4/Makefile

```diff
@@ -0,0 +1,39 @@
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+JSLIB ?= targets/liteos-stm32f4/lib
+
+# path to the LiteOS base directory
+SOURCE_ROOT ?= $(CURDIR)/../LiteOS_Kernel
+# path to the JERRYSCRIPT directory
+JERRYDIR ?= $(CURDIR)
+
+# path to the application directory
+APPDIR ?= $(JERRYDIR)/targets/liteos-stm32f4/source
+
+# path to the binary directory
+BINDIR ?= $(JERRYDIR)/targets/liteos-stm32f4/bin/
+
+
+# Change this to 0 show compiler invocation lines by default:
+QUIET ?= 1
+
+INCLUDES ?= -I$(JERRYDIR)/jerry-core/include
+
+
+# Add the jerry lib
+LDFLAGS ?= ${INCLUDES} -L $(JSLIB) -ljerrycore 
```

Comment:
It's still the *LiteOS_Kernel* that sets up JerryScript-specific things. Excerpt from the latest `LiteOS_Kernel/projects/STM32F429I_DISCO_GCC/Makefile`:

```
ifneq (, $(LITEOSBASE))
SRC += ${wildcard $(TOP_DIR)/../jerryscript/targets/liteos-stm32f4/source/*.c}
else
SRC += ${wildcard $(TOP_DIR)/user/*.c}
endif

ifneq  (, $(LITEOSBASE)) 
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/include
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/ecma/base
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/jrt
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/ecma/operations
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/lit
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/jmem
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/ecma/builtin-objects
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/vm
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/jcontext
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/parser/js
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/debugger
CFLAGS += -I $(TOP_DIR)/../jerryscript/jerry-core/parser/regexp
CFLAGS += -I $(TOP_DIR)/../jerryscript/targets/liteos-stm32f4/source
endif

$(TARGET_ELF):$(OBJ) $(ASMOBJ) 
	$(CC) $(CFLAGS) $(LDFLAGS) $^  -o $@  $(JS_CFLAGS)
```

You'll have to find a way to break out JerryScript references from kernel makefiles. Includes can be added to CFLAGS (not LDFLAGS!), and sources might be added to SRC (if it's not set/overwritten with := later).